modules/sr/robot/robot.py

```diff
@@ -152,38 +151,3 @@ def sleep(self, secs: float) -> None:
         # cleanup if Webots tells us the simulation is terminating. When webots
         # kills the process all the proper tidyup will happen anyway.
         self.webots_step_and_should_continue(duration_ms)
-
-
-class AutomaticTimestepRobot(ManualTimestepRobot):
-    """
-    Robot class which preserves the original automatic time-advancing behaviour.
-
-    This class launches a background thread which advances the timestep in a
-    tight loop. This is somewhat more convenient to program against because it
-    does not rely on the `sleep` method being called in order for time to
-    advance. However as a result the timestep is considerably less predictable
-    which can result in unexpected robot behaviours.
-
-    The `sleep` method of this class is still available and is thread-safe.
-    """
-
-    def init(self) -> None:
-        self.webots_init()
-        super().init()
-
-    def webots_init(self) -> None:
-        # Create a thread which will advance time in the background, so that the
-        # competitors' code can ignore the fact that it is actually running in a
-        # simulation.
-        t = Thread(
-            target=self.webot_run_robot,
-            # Ensure our background thread alone won't keep the controller
-            # process runnnig.
-            daemon=True,
-        )
-        t.start()
-        time.sleep(self._timestep / 1000)
-
-    def webot_run_robot(self):
-        while self.webots_step_and_should_continue(self._timestep):
```

Comment:
I wonder whether we should remove the `_and_should_continue` functionality?

Reply:
I can see it potentially being useful for competitors. I have fixed a related comment in 89ea239 though.